src/rust/engine/fs/src/lib.rs

```diff
@@ -648,6 +731,52 @@ impl Vfs<String> for DigestTrie {
         )))
     }
 
+    async fn path_metadata(&self, path: PathBuf) -> Result<Option<PathMetadata>, String> {
```

Comment:
I'm not fully across this part of the engine... is a problem that these ones are mostly empty? It seems quite different to the "real FS" one above, and I'm not sure of the consequences.

Reply:
This code is filling in the `PathMetadata` from a `Digest`. A `Digest` of course appears like a filesystem tree but doesn't have the complete set of data which a filesystem would have.

That said, there are some extensions to REAPI which allow storing UNIX permission modes and such, but Pants doss not support those extensions currently. (And it would be a separate issue from this one whether it should or not. But if it did, we could use the extensions to fill in more data here.)